security/auditing.rst

```diff
@@ -236,3 +236,125 @@ The table below contains some elements you can find in the generated AuditEvents
 | Example ($erase operation)  | :download:`download <../_static/files/audit-event-examples/R3_erase_operation.json>`  | :download:`download <../_static/files/audit-event-examples/R4_erase_operation.json>`  | :download:`download <../_static/files/audit-event-examples/R5_erase_operation.json>`  |
 +-----------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+
+
+.. _audit_event_customization:
+
+AuditEvent customization
+------------------------
+
+If you need to include additional information in the standard AuditEvents, you can do that with a custom plugin. 
+
+To implement such a plugin, it is helpful to understand how AuditEvents get created in Firely Server. Whenever the server receives an incoming HTTP request, a middleware registered in ``AuditEventConfiguration`` first passes it transparently to the downstream handlers. Then, when the original requests get handled, the audit middleware creates another artificial request and passes it down the stream again. This time, the request contains a creation operation with the AuditEvent as a payload. Like any other request in Firely Server, this request can be intercepted and changed using a pre-handler before it continues down the pipeline until ``CreateOperationMiddleware`` handles it. The order of the customization plugin should be greater than ``3170`` and less than ``4420``.
+  
+
+.. @startuml
+
+.. title "AuditEvent creation process"
+
+.. participant "AuditEventMiddleware (order 3170)" as aem
+.. participant "..." as hOther
+.. participant "AuditEventCustomizationMiddleware" as aecm
+.. participant "..." as hOther2
+.. participant "CreateOperationMiddleware (order 4420)" as hCreate
+
+.. -> aem: original HTTP request
+.. activate aem
+.. aem -> hOther: original HTTP request
+.. activate hOther
+.. hOther --> aem
+.. deactivate hOther
+
+.. aem -> hOther: AuditEvent creation \nrequest context
+.. activate hOther
+.. hOther -> aecm: AuditEvent creation \nrequest context
+.. activate aecm
+.. aecm -> aecm: Customize payload
+.. activate aecm
+.. deactivate aecm
+.. aecm -> hOther2: AuditEvent creation \nrequest context
+.. activate hOther2
+.. hOther2 -> hCreate
+.. activate hCreate
+.. hCreate --> hOther2
+.. deactivate hCreate
+.. hOther2 --> aecm
+.. deactivate hOther2
+.. aecm --> hOther
+.. deactivate aecm
+.. hOther --> aem
+.. deactivate hOther
+.. <-- aem
+.. deactivate aem
+
+.. @enduml
+
+.. image:: ../../_static/images/auditing/AuditEvent-customization.svg
+   :width: 800
+
+
+See an example plugin below. This plugin captures all the token claims from the original request and then includes those claims into the AuditEvent.
```

Comment:
I would highlight that you need to work with SourceNodes at this level. Either you need to continue to use SourceNodes (recommended) or you need to write some transform it to a POCO. Maybe you could link to the relevant classes here?